stumpy/core.py

```diff
@@ -962,30 +962,39 @@ def compute_mean_std(T, m):
     # "f8(i8, f8, f8, f8, f8, f8)",
     fastmath=True
 )
-def _calculate_squared_distance(m, QT, μ_Q, σ_Q, M_T, Σ_T):
+def _calculate_squared_distance(
+    m, QT, μ_Q, σ_Q, M_T, Σ_T, Q_subseq_isconstant, T_subseq_isconstant
+):
     """
     Compute a single squared distance given all scalar inputs. This function serves as
     the single source of truth for how all distances should be calculated.
+    In this function, Q and T (in `QT`) are two subsequences, each with length `m`.
```

Comment:
I just added it for clarification. Because, I am usually inclined to think that `Q` contains one subeq of length `m`, and `T` contains a **bunch** of subseqs. So, when I see the term "single" in  `Compute a single squared distance given all scalar inputs`, it creates a small confusion in my head. If I continue reading the docstring and read the definition of parameters, I can realize that `Q` and `T`, each,  is just **one** subsequence. So, I thought adding such explanation in the beginning of the docstring might help with clarifying the purpose of this function. When I say length `m`, that means each of the two time series `Q`  and `T` contains only one subseq.

What do you think?

Reply:
But, again, my issue is that there are no parameters `Q` and `T` anywhere in this function. There is only `QT`, `Q_subseq_isconstant`, and `T_subseq_isconstant`. So, the docstring that you've added has not defined what `Q` or `T` is. I think it is slightly better to start with `QT`:

```
In this function, `QT` is pre-computed from two subsequences, `Q` and `T`, each with length `m`.
```

This way, it sounds like you are explaining `QT`. In your original description, it sounds like you are trying to explain what `Q` and `T` are. Also, I would simply move this to the docstring for `QT`?